### PR TITLE
Return sorted messages and fixed http status code on exceptions

### PR DESCRIPTION
**Pull request by Vekeryk**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

**Review by marianben — APPROVED** [review]

**Review by marianben — APPROVED** [review]

File: src/main/java/com/softserve/teachua/repository/MessageRepository.java [hunk 1]
```diff
@@ -13,7 +13,7 @@
 @Repository
 public interface MessageRepository extends JpaRepository<Message, Long> {
 
-    Optional<List<Message>> findAllBySenderId(Long id);
+    Optional<List<Message>> findAllBySenderIdOrderByDateDesc(Long id);
```
Comment (marianben) [inline src/main/java/com/softserve/teachua/repository/MessageRepository.java:16]: Why did you change the name of the method? (Only for readability or did you have other reasons for it) And did you check whether there are no connections with this method?

**Review by marianben — APPROVED** [review]

File: src/main/java/com/softserve/teachua/repository/MessageRepository.java (hunk 1 above)
Comment (Vekeryk) [reply src/main/java/com/softserve/teachua/repository/MessageRepository.java:16]: Messages on user profile page were sorted by js on fronted side. So I return already sorted messages by date from API (unsorted makes no sense) by using JPA repository method.

**Review by softservedata — APPROVED** [review]
